scipy/linalg/_matfuncs_inv_ssq.py

```diff
@@ -298,12 +299,12 @@ def _logm_superdiag_entry(l1, l2, t12):
     """
     if l1 == l2:
         f12 = t12 / l1
-    elif abs(l1) < abs(l2) / 2 or abs(l2) < abs(l1) / 2:
+    elif abs(l2 - l1) > abs(l1 + l2) / 2:
         f12 = t12 * (np.log(l2) - np.log(l1)) / (l2 - l1)
     else:
         z = (l2 - l1) / (l2 + l1)
         ua = _unwindk(np.log(l2) - np.log(l1))
-        ub = _unwindk(np.log(1+z) - np.log(1-z))
+        ub = _unwindk(np.log1p(z) - np.log1p(z))
```

Comment:
@pv I've just removed that line, because I think `ub` was always zero regardless of the inputs, with or without the typo I introduced in this PR.  This is discussed in the reference near "it is important to note that atanh can be defined in a different way" around equations 11.24 -- 11.27 in Higham's "Functions of Matrices: Theory and Computation".